config/config_helpers.go

```diff
@@ -28,19 +52,46 @@ type EnvVar struct {
 func ResolveTerragruntConfigString(terragruntConfigString string, include *IncludeConfig, terragruntOptions *options.TerragruntOptions) (resolved string, finalErr error) {
 	// The function we pass to ReplaceAllStringFunc cannot return an error, so we have to use named error
 	// parameters to capture such errors.
-	resolved = INTERPOLATION_SYNTAX_REGEX.ReplaceAllStringFunc(terragruntConfigString, func(str string) string {
+
+	resolved = INTERPOLATION_SYNTAX_REGEX_SINGLE.ReplaceAllStringFunc(terragruntConfigString, func(str string) string {
+		// We do a special treatment for function returning an array of values because they need
+		// to be declared like that "${function()" and the result must get rid of the surrounding quote
+		// to return a real array of values.
+		internalStr := str[1 : len(str)-1] // We get rid of the surrounding quote
+		out, err := resolveTerragruntInterpolation(internalStr, include, terragruntOptions)
+		if err != nil {
+			finalErr = err
+		}
+
+		switch out := out.(type) {
+		case []string:
+			return util.ListToHCLArray(out)
+		}
```

Comment:
There should be two `case` statements here: the one for `[]string` you already have and a new one for `string` that returns the string unchanged. There should also be a `default` that sets `finalErr` to an error message indicating that an unsupported type was returned and using `reflect.TypeOf` to display what that type was.